### PR TITLE
Change tests and compilation to run on Net6

### PR DESCRIPTION
**Pull request by wallymathieu**
- [x] testDefaultInferenceForReturnFrom

### CODE REVIEW

File: tests/FSharpPlus.Tests/Task.fs [hunk 1]
```diff
@@ -828,7 +828,7 @@ module Task =
 
         // no need to call this, we just want to check that it compiles w/o warnings
         let testDefaultInferenceForReturnFrom() =
-            let t: Task<_> = monad' { return Some "x" }
+            let t: Task<string option> = monad' { return Some "x" }
```
Comment (wallymathieu) [inline tests/FSharpPlus.Tests/Task.fs:831]: This pops out

File: .config/dotnet-tools.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
       ]
     },
     "fable": {
-      "version": "3.2.6",
+      "version": "3.6.3",
```
Comment (wallymathieu) [inline .config/dotnet-tools.json:18]: I tried bumping to 3.7, but got some compilation errors

File: tests/FSharpPlus.Tests/Task.fs (hunk 1 above)
Comment (gusty) [reply tests/FSharpPlus.Tests/Task.fs:831]: This is a regression. I posted a minimized repro here: https://github.com/dotnet/fsharp/issues/12929

**Review by gusty — APPROVED** [review]
Looks good. Let's put something more descriptive than just Net6 and merge it.

**Comment by wallymathieu** [comment]
Any ideas what we should call it?

**Comment by gusty** [comment]
That's perfect, very descriptive and concise.

**Pull request merged**